gst/nnstreamer/nnstreamer_conf.c

```diff
@@ -254,12 +254,19 @@ _fill_in_vstr (gchar *** fullpath_vstr, gchar *** basename_vstr,
 {
   GSList *lstF = NULL, *lstB = NULL;
   vstr_helper vstrF, vstrB;
-  guint i, counter;
+  guint i, j, counter;
 
   counter = 0;
   for (i = 0; i < CONF_SOURCE_END; i++) {
     if (searchpath[i]) {
-      _get_filenames (type, searchpath[i], &lstF, &lstB, &counter);
+      /* skip duplicated paths */
+      for (j = i + 1; j < CONF_SOURCE_END; j++) {
+        if (searchpath[j] && !g_strcmp0 (searchpath[i], searchpath[j])) {
```

Comment:
This logic will avoid duplicated paths but not equivalent paths.
I am not sure if this can happen but unnecessary `. or ..` in the paths can make equivalent paths which aren't equal in string format. How about using `g_file_new_for_path()` to create a GFile and then check equivalence of path with `f_file_equal()`?

Reply:
Yes, it could be better, but it may require extra dependency (e.g., `gio`) and I think it's a quite overkill to consider such rare cases.

Reply:
ok.